specification.markdown

```diff
@@ -228,6 +210,95 @@ VINT\_WIDTH | VINT\_MARKER | VINT\_DATA     | Element Data Size Status
         For example -2 can be coded as 0xFFFFFFFFFFFFFE or 0xFFFE or
         0xFE and 5 can be coded 0x000000000005 or 0x0005 or 0x05.
 
+## EBML Element Types
+
+Each defined EBML Element MUST have a declared Element Type. The Element Type defines a concept for storing data that may be constrained by length, endianness, and purpose.
+
+Element Data Type:   Signed Integer
+
+    Endianness:     Big-endian
+    Length:         A Signed Integer Element MUST at least 1 octet in
```

Comment:
I agree 0-length elements should be allowed.

And avoiding special cases for dates, even if it's never used. In libebml that would require adding code because when reading an element we read he ID, the length and we verify it's legit (otherwise the data are likely bogus and we don't use them). So we'd have to check if a date element has a size of 0. And in the end, why do we want to reject it ? (a date in libebml is an EBML signed integer on which we can read/write a date, I'd assume most implementations would do the same).